The terraform-provider-ibm_v1.64.0 plugin crashed!

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

terraform 1.5.7
ibm provider version 1.64.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_scc_profile_attachment

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

See https://github.com/terraform-ibm-modules/terraform-ibm-scc/pull/65
Logs at https://github.com/terraform-ibm-modules/terraform-ibm-scc/actions/runs/8542182407/job/23403425951

### Debug Output

```
│ Error: Plugin did not respond
│ 
│   with module.create_profile_attachment.ibm_scc_profile_attachment.scc_profile_attachment,
│   on ../../modules/attachment/main.tf line 40, in resource "ibm_scc_profile_attachment" "scc_profile_attachment":
│   40: resource "ibm_scc_profile_attachment" "scc_profile_attachment" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-ibm_v1.64.0 plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 349 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentMapToAttachmentsPrototype(0x409d100?)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:637 +0x9d4
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentMapToAttachmentPrototype(0x409d100?)
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:841 +0x13b
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc.resourceIbmSccProfileAttachmentCreate({0x5153d98, 0xc001f57e60}, 0x0?, {0x47eb140?, 0xc000ae0a80})
	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/scc/resource_ibm_scc_profile_attachment.go:310 +0x71f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc001423180, {0x5153dd0, 0xc001e137d0}, 0xd?, {0x47eb140, 0xc000ae0a80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:778 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001423180, {0x5153dd0, 0xc001e137d0}, 0xc0021ccc30, 0xc001e2a780, {0x47eb140, 0xc000ae0a80})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:909 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002ed488, {0x5153dd0?, 0xc001e136b0?}, 0xc001e34140)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000a62640, {0x5153dd0?, 0xc001e12ea0?}, 0xc0002c07e0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x46660e0?, 0xc000a62640}, {0x5153dd0, 0xc001e12ea0}, 0xc0002c0770, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000a521e0, {0x515ebe0, 0xc00097c4e0}, 0xc001e2c900, 0xc0014bcd50, 0x717ee98, 0x0)
	google.golang.org/grpc@v1.57.1/server.go:1358 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000a521e0, {0x515ebe0, 0xc00097c4e0}, 0xc001e2c900, 0x0)
	google.golang.org/grpc@v1.57.1/server.go:1735 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.1/server.go:970 +0xca
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.1/server.go:981 +0x15c

Error: The terraform-provider-ibm_v1.64.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/service/scc/resource_ibm_scc_profile_attachment.go
+++ b/ibm/service/scc/resource_ibm_scc_profile_attachment.go
@@ -437,7 +437,7 @@ func resourceIbmSccProfileAttachmentRead(context context.Context, d *schema.Reso
 		}
 	}
 	if !core.IsNil(attachmentItem.AttachmentParameters) {
-		attachmentParameters := d.Get("attachment_parameters").(*schema.Set)
+		attachmentParameters := &schema.Set{F: hashAttachmentParameters}
 		for _, attachmentParametersItem := range attachmentItem.AttachmentParameters {
 			attachmentParametersItemMap, err := resourceIbmSccProfileAttachmentAttachmentParameterPrototypeToMap(&attachmentParametersItem)
 			if err != nil {

--- a/ibm/service/scc/resource_ibm_scc_profile_attachment.go
+++ b/ibm/service/scc/resource_ibm_scc_profile_attachment.go
@@ -7,9 +7,6 @@ import (
 	"context"
 	"fmt"
 	"log"
-	"math/big"
-	"strconv"
-	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -168,40 +165,8 @@ func ResourceIbmSccProfileAttachment() *schema.Resource {
 				Type:        schema.TypeSet,
 				Optional:    true,
 				Description: "The profile parameters for the attachment.",
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"assessment_type": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							Description: "The type of the implementation.",
-						},
-						"assessment_id": {
-							Type:        schema.TypeString,
-							Required:    true,
-							Description: "The implementation ID of the parameter.",
-						},
-						"parameter_name": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							Description: "The parameter name.",
-						},
-						"parameter_value": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							Description: "The value of the parameter.",
-						},
-						"parameter_display_name": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							Description: "The parameter display name.",
-						},
-						"parameter_type": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							Description: "The parameter type.",
-						},
-					},
-				},
+				Elem:        schemaAttachmentParameters(),
+				Set:         hashAttachmentParameters,
 			},
 			"last_scan": {
 				Type:        schema.TypeList,
@@ -269,6 +234,56 @@ func ResourceIbmSccProfileAttachmentValidator() *validate.ResourceValidator {
 	return &resourceValidator
 }
 
+// hashAttachmentParameters will determine how to hash the AttachmentParameters schema.Resource
+// It uses the 'assessment_id' in order to determine the difference.
+func hashAttachmentParameters(v interface{}) int {
+	if v == nil {
+		return 0
+	}
+	m := v.(map[string]interface{})
+	id := (m["assessment_id"]).(string)
+	return schema.HashString(id)
+}
+
+// schemaAttachmentParameters returns a *schema.Resource for AttachmentParameters
+func schemaAttachmentParameters() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"assessment_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The implementation ID of the parameter.",
+			},
+			"assessment_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "automated",
+				Description: "The type of the implementation.",
+			},
+			"parameter_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The parameter name.",
+			},
+			"parameter_display_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The parameter display name.",
+			},
+			"parameter_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The parameter type.",
+			},
+			"parameter_value": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The value of the parameter.",
+			},
+		},
+	}
+}
+
 func resourceIbmSccProfileAttachmentCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	securityandcompliancecenterapiClient, err := meta.(conns.ClientSession).SecurityAndComplianceCenterV3()
 	if err != nil {
@@ -279,6 +294,7 @@ func resourceIbmSccProfileAttachmentCreate(context context.Context, d *schema.Re
 	createAttachmentOptions := &securityandcompliancecenterapiv3.CreateAttachmentOptions{}
 	instance_id := d.Get("instance_id").(string)
 	bodyModelMap["instance_id"] = instance_id
+
 	if _, ok := d.GetOk("profile_id"); ok {
 		bodyModelMap["profile_id"] = d.Get("profile_id")
 	}
@@ -288,11 +304,10 @@ func resourceIbmSccProfileAttachmentCreate(context context.Context, d *schema.Re
 	if _, ok := d.GetOk("scope"); ok {
 		bodyModelMap["scope"] = d.Get("scope")
 	}
-	// manual chang
+
+	// manual change
 	if _, ok := d.GetOk("attachment_parameters"); ok {
 		bodyModelMap["attachment_parameters"] = d.Get("attachment_parameters")
-	} else {
-		bodyModelMap["attachment_parameters"] = []interface{}{}
 	}
 	if _, ok := d.GetOk("notifications"); ok {
 		bodyModelMap["notifications"] = d.Get("notifications")
@@ -322,22 +337,6 @@ func resourceIbmSccProfileAttachmentCreate(context context.Context, d *schema.Re
 	d.SetId(fmt.Sprintf("%s/%s/%s", instance_id, *createAttachmentOptions.ProfileID, *attachmentPrototype.Attachments[0].ID))
 
 	return resourceIbmSccProfileAttachmentRead(context, d, meta)
-}
-
-func cmpAttachParamSetFunc(v interface{}) int {
-	if v == nil {
-		return 0
-	}
-	m := v.(map[string]interface{})
-	id := (m["assessment_id"]).(*string)
-	assId := (*id)[5:18]
-	var i big.Int
-	i.SetString(strings.Replace(assId, "-", "", 4), 16)
-	val, err := strconv.Atoi(i.String())
-	if err != nil {
-		log.Printf("[ERROR] Setting the Parameters of the Profile Attachment failed %s\n", err)
-	}
-	return val
 }
 
 func resourceIbmSccProfileAttachmentRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
@@ -438,15 +437,14 @@ func resourceIbmSccProfileAttachmentRead(context context.Context, d *schema.Reso
 		}
 	}
 	if !core.IsNil(attachmentItem.AttachmentParameters) {
-		attachmentParametersList := []interface{}{}
+		attachmentParameters := d.Get("attachment_parameters").(*schema.Set)
 		for _, attachmentParametersItem := range attachmentItem.AttachmentParameters {
 			attachmentParametersItemMap, err := resourceIbmSccProfileAttachmentAttachmentParameterPrototypeToMap(&attachmentParametersItem)
 			if err != nil {
 				return diag.FromErr(err)
 			}
-			attachmentParametersList = append(attachmentParametersList, attachmentParametersItemMap)
+			attachmentParameters.Add(attachmentParametersItemMap)
 		}
-		attachmentParameters := schema.NewSet(cmpAttachParamSetFunc, attachmentParametersList)
 		if err = d.Set("attachment_parameters", attachmentParameters); err != nil {
 			return diag.FromErr(fmt.Errorf("Error setting attachment_parameters: %s", err))
 		}
@@ -523,12 +521,17 @@ func resourceIbmSccProfileAttachmentUpdate(context context.Context, d *schema.Re
 		hasChange = true
 	}
 
-	if d.HasChange("attachment_item") {
-		attachmentItem, err := resourceIbmSccProfileAttachmentMapToAttachmentItem(d.Get("attachment_item.0").(map[string]interface{}))
-		if err != nil {
-			return diag.FromErr(err)
+	if d.HasChange("attachment_parameters") {
+		attachmentItems := d.Get("attachment_parameters")
+		attachmentParameters := []securityandcompliancecenterapiv3.AttachmentParameterPrototype{}
+		for _, attachmentParametersItem := range attachmentItems.(*schema.Set).List() {
+			attachmentParametersItemModel, err := resourceIbmSccProfileAttachmentMapToAttachmentParameterPrototype(attachmentParametersItem.(map[string]interface{}))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			attachmentParameters = append(attachmentParameters, *attachmentParametersItemModel)
 		}
-		replaceProfileAttachmentOptions.SetAttachmentID(*attachmentItem.ID)
+		replaceProfileAttachmentOptions.SetAttachmentParameters(attachmentParameters)
 		hasChange = true
 	}
 
@@ -634,13 +637,15 @@ func resourceIbmSccProfileAttachmentMapToAttachmentsPrototype(modelMap map[strin
 		model.Notifications = NotificationsModel
 	}
 	attachmentParameters := []securityandcompliancecenterapiv3.AttachmentParameterPrototype{}
-	for _, attachmentParametersItem := range modelMap["attachment_parameters"].([]interface{}) {
-		if attachmentParametersItem != nil {
-			attachmentParametersItemModel, err := resourceIbmSccProfileAttachmentMapToAttachmentParameterPrototype(attachmentParametersItem.(map[string]interface{}))
-			if err != nil {
-				return model, err
+	if modelMap["attachment_parameters"] != nil {
+		for _, attachmentParametersItem := range modelMap["attachment_parameters"].(*schema.Set).List() {
+			if attachmentParametersItem != nil {
+				attachmentParametersItemModel, err := resourceIbmSccProfileAttachmentMapToAttachmentParameterPrototype(attachmentParametersItem.(map[string]interface{}))
+				if err != nil {
+					return model, err
+				}
+				attachmentParameters = append(attachmentParameters, *attachmentParametersItemModel)
 			}
-			attachmentParameters = append(attachmentParameters, *attachmentParametersItemModel)
 		}
 	}
 	model.AttachmentParameters = attachmentParameters
@@ -834,6 +839,7 @@ func resourceIbmSccProfileAttachmentMapToLastScan(modelMap map[string]interface{
 
 func resourceIbmSccProfileAttachmentMapToAttachmentPrototype(modelMap map[string]interface{}) (*securityandcompliancecenterapiv3.CreateAttachmentOptions, error) {
 	model := &securityandcompliancecenterapiv3.CreateAttachmentOptions{}
+	model.SetInstanceID(modelMap["instance_id"].(string))
 	if modelMap["profile_id"] != nil && modelMap["profile_id"].(string) != "" {
 		model.ProfileID = core.StringPtr(modelMap["profile_id"].(string))
 	}
@@ -844,7 +850,6 @@ func resourceIbmSccProfileAttachmentMapToAttachmentPrototype(modelMap map[string
 	}
 	attachments = append(attachments, *attachmentsItemModel)
 	model.Attachments = attachments
-	model.SetInstanceID(modelMap["instance_id"].(string))
 	return model, nil
 }
 
@@ -899,22 +904,22 @@ func resourceIbmSccProfileAttachmentFailedControlsToMap(model *securityandcompli
 func resourceIbmSccProfileAttachmentAttachmentParameterPrototypeToMap(model *securityandcompliancecenterapiv3.AttachmentParameterPrototype) (map[string]interface{}, error) {
 	modelMap := make(map[string]interface{})
 	if model.AssessmentType != nil {
-		modelMap["assessment_type"] = model.AssessmentType
+		modelMap["assessment_type"] = flex.StringValue(model.AssessmentType)
 	}
 	if model.AssessmentID != nil {
-		modelMap["assessment_id"] = model.AssessmentID
+		modelMap["assessment_id"] = flex.StringValue(model.AssessmentID)
 	}
 	if model.ParameterName != nil {
-		modelMap["parameter_name"] = model.ParameterName
+		modelMap["parameter_name"] = flex.StringValue(model.ParameterName)
 	}
 	if model.ParameterValue != nil {
-		modelMap["parameter_value"] = model.ParameterValue
+		modelMap["parameter_value"] = flex.StringValue(model.ParameterValue)
 	}
 	if model.ParameterDisplayName != nil {
-		modelMap["parameter_display_name"] = model.ParameterDisplayName
+		modelMap["parameter_display_name"] = flex.StringValue(model.ParameterDisplayName)
 	}
 	if model.ParameterType != nil {
-		modelMap["parameter_type"] = model.ParameterType
+		modelMap["parameter_type"] = flex.StringValue(model.ParameterType)
 	}
 	return modelMap, nil
 }

--- a/ibm/service/scc/resource_ibm_scc_profile_attachment_test.go
+++ b/ibm/service/scc/resource_ibm_scc_profile_attachment_test.go
@@ -53,6 +53,10 @@ func TestAccIbmSccProfileAttachmentAllArgs(t *testing.T) {
 				Check:  resource.ComposeAggregateTestCheckFunc(),
 			},
 			resource.TestStep{
+				Config: testAccCheckIbmSccProfileAttachmentConfigChange(acc.SccInstanceID),
+				Check:  resource.ComposeAggregateTestCheckFunc(),
+			},
+			resource.TestStep{
 				ResourceName:      "ibm_scc_profile_attachment.scc_profile_attachment_instance",
 				ImportState:       true,
 				ImportStateVerify: true,
@@ -148,70 +152,30 @@ func testAccCheckIbmSccProfileAttachmentConfigBasic(instanceID string) string {
 
 func testAccCheckIbmSccProfileAttachmentConfig(instanceID string) string {
 	return fmt.Sprintf(`
-		resource "ibm_scc_control_library" "scc_control_library_instance" {
-			instance_id = "%s"
-			control_library_name = "control_library_name"
-			control_library_description = "control_library_description"
-			control_library_type = "custom"
-			version_group_label = "03354ab4-03be-41c0-a469-826fc0262e78"
-			latest = true
-			controls {
-				control_name = "control-name"
-				control_id = "1fa45e17-9322-4e6c-bbd6-1c51db08e790"
-				control_description = "control_description"
-				control_category = "control_category"
-				control_tags = [ "control_tags" ]
-				control_specifications {
-					control_specification_id = "f3517159-889e-4781-819a-89d89b747c85"
-					responsibility = "user"
-					component_id = "f3517159-889e-4781-819a-89d89b747c85"
-					component_name = "f3517159-889e-4781-819a-89d89b747c85"
-					environment = "environment"
-					control_specification_description = "control_specification_description"
-					assessments {
-						assessment_id = "rule-a637949b-7e51-46c4-afd4-b96619001bf1"
-						assessment_method = "ibm-cloud-rule"
-						assessment_type = "automated"
-						assessment_description = "assessment_description"
-						parameters {
-							parameter_display_name = "Sign out due to inactivity in seconds"
-                            parameter_name         = "session_invalidation_in_seconds"
-							parameter_type = "numeric"
-						}
-					}
-				}
-				control_docs {
-					control_docs_id = "control_docs_id"
-					control_docs_type = "control_docs_type"
-				}
-				control_requirement = true
-				status = "enabled"
-			}
-		}
+	locals {
+		scc_profiles_map = tomap(merge([
+			for i , cl in data.ibm_scc_profiles.scc_profiles.profiles :
+			{(cl.profile_name) = "${cl.id}"}  if cl.latest == true && cl.profile_type == "predefined"
+		]...))
+	}
 
-		resource "ibm_scc_profile" "scc_profile_instance" {
-			instance_id = resource.ibm_scc_control_library.scc_control_library_instance.instance_id
-			profile_name = "profile_name"
-			profile_description = "profile_description"
-			profile_type = "custom"
-			controls {
-				control_library_id = resource.ibm_scc_control_library.scc_control_library_instance.control_library_id
-				control_id = resource.ibm_scc_control_library.scc_control_library_instance.controls[0].control_id
-			}
-			default_parameters {
-			}
-		}
+	data "ibm_scc_profiles" "scc_profiles" {
+		instance_id = "%s"
+	}
 
-		resource "ibm_scc_profile_attachment" "scc_profile_attachment_instance" {
-			instance_id = resource.ibm_scc_control_library.scc_control_library_instance.instance_id
-			profile_id = ibm_scc_profile.scc_profile_instance.profile_id
-			name = "profile_attachment_name"
-			description = "scc_profile_attachment_description"
+    data "ibm_iam_account_settings" "iam_account_settings" {
+    }
+
+    resource "ibm_scc_profile_attachment" "scc_profile_attachment_instance" {
+		instance_id = "%s"
+		profile_id = local.scc_profiles_map["CIS IBM Cloud Foundations Benchmark"]
+		name = "profile_attachment_name"
+		description = "scc_profile_attachment_description"
 			scope {
 				environment = "ibm-cloud"	
 				properties {
 					name = "scope_id"
-					value = resource.ibm_scc_control_library.scc_control_library_instance.account_id
+					value = data.ibm_iam_account_settings.iam_account_settings.account_id
 				}
 				properties {
 					name = "scope_type"
@@ -219,7 +183,7 @@ func testAccCheckIbmSccProfileAttachmentConfig(instanceID string) string {
 				}
 			}
 			schedule = "every_30_days"
-			status = "enabled"
+			status = "disabled"
 			notifications {
 				enabled = false
 				controls {
@@ -227,8 +191,133 @@ func testAccCheckIbmSccProfileAttachmentConfig(instanceID string) string {
 					threshold_limit = 14
 				}
 			}
+		attachment_parameters {
+			parameter_value = "['1.2', '1.3']"
+			assessment_id = "rule-e16fcfea-fe21-4d30-a721-423611481fea"
+        	assessment_type = "automated"
+			parameter_display_name = "IBM Cloud Internet Services TLS version"
+			parameter_name = "tls_version"
+			parameter_type = "string_list"
 		}
-	`, instanceID)
+		attachment_parameters {
+			parameter_value = "22"
+			assessment_id = "rule-f9137be8-2490-4afb-8cd5-a201cb167eb2"
+			assessment_type = "automated"
+			parameter_display_name = "Network ACL rule for allowed IPs to SSH port"
+			parameter_name = "ssh_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "3389"
+			assessment_id = "rule-9653d2c7-6290-4128-a5a3-65487ba40370"
+			assessment_type = "automated"
+			parameter_display_name = "Security group rule RDP allow port number"
+			parameter_name = "rdp_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "22"
+			assessment_id = "rule-7c5f6385-67e4-4edf-bec8-c722558b2dec"
+			assessment_type = "automated"
+			parameter_display_name = "Security group rule SSH allow port number"
+			parameter_name = "ssh_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "3389"
+			assessment_id = "rule-f1e80ee7-88d5-4bf2-b42f-c863bb24601c"
+			assessment_type = "automated"
+			parameter_display_name = "Disallowed IPs for ingress to RDP port"
+			parameter_name = "rdp_port"
+			parameter_type = "numeric"
+		}
+	}
+	`, instanceID, instanceID)
+}
+
+func testAccCheckIbmSccProfileAttachmentConfigChange(instanceID string) string {
+	return fmt.Sprintf(`
+	locals {
+		scc_profiles_map = tomap(merge([
+			for i , cl in data.ibm_scc_profiles.scc_profiles.profiles :
+			{(cl.profile_name) = "${cl.id}"}  if cl.latest == true && cl.profile_type == "predefined"
+		]...))
+	}
+
+	data "ibm_scc_profiles" "scc_profiles" {
+		instance_id = "%s"
+	}
+
+    data "ibm_iam_account_settings" "iam_account_settings" {
+    }
+
+    resource "ibm_scc_profile_attachment" "scc_profile_attachment_instance" {
+		instance_id = "%s"
+		profile_id = local.scc_profiles_map["CIS IBM Cloud Foundations Benchmark"]
+		name = "profile_attachment_name"
+		description = "scc_profile_attachment_description"
+			scope {
+				environment = "ibm-cloud"	
+				properties {
+					name = "scope_id"
+					value = data.ibm_iam_account_settings.iam_account_settings.account_id
+				}
+				properties {
+					name = "scope_type"
+					value = "account"
+				}
+			}
+			schedule = "every_30_days"
+			status = "disabled"
+			notifications {
+				enabled = false
+				controls {
+					failed_control_ids = []
+					threshold_limit = 14
+				}
+			}
+		attachment_parameters {
+			parameter_value = "['1.2', '1.3']"
+			assessment_id = "rule-e16fcfea-fe21-4d30-a721-423611481fea"
+        	assessment_type = "automated"
+			parameter_display_name = "IBM Cloud Internet Services TLS version"
+			parameter_name = "tls_version"
+			parameter_type = "string_list"
+		}
+		attachment_parameters {
+			parameter_value = "22"
+			assessment_id = "rule-f9137be8-2490-4afb-8cd5-a201cb167eb2"
+			assessment_type = "automated"
+			parameter_display_name = "Network ACL rule for allowed IPs to SSH port"
+			parameter_name = "ssh_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "22"
+			assessment_id = "rule-7c5f6385-67e4-4edf-bec8-c722558b2dec"
+			assessment_type = "automated"
+			parameter_display_name = "Security group rule SSH allow port number"
+			parameter_name = "ssh_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "3389"
+			assessment_id = "rule-9653d2c7-6290-4128-a5a3-65487ba40370"
+			assessment_type = "automated"
+			parameter_display_name = "Security group rule RDP allow port number"
+			parameter_name = "rdp_port"
+			parameter_type = "numeric"
+		}
+		attachment_parameters {
+			parameter_value = "3389"
+			assessment_id = "rule-f1e80ee7-88d5-4bf2-b42f-c863bb24601c"
+			assessment_type = "automated"
+			parameter_display_name = "Disallowed IPs for ingress to RDP port"
+			parameter_name = "rdp_port"
+			parameter_type = "numeric"
+		}
+	}
+	`, instanceID, instanceID)
 }
 
 func testAccCheckIbmSccProfileAttachmentExists(n string, obj securityandcompliancecenterapiv3.AttachmentItem) resource.TestCheckFunc {

--- a/website/docs/r/scc_profile_attachment.html.markdown
+++ b/website/docs/r/scc_profile_attachment.html.markdown
@@ -40,6 +40,13 @@ resource "ibm_scc_profile_attachment" "scc_profile_attachment_instance" {
       threshold_limit = 14
     }
   }
+  attachment_parameters {
+    parameter_value = "22"
+    assessment_id = "rule-this-is-a-fake-ruleid"
+    parameter_display_name = "Network ACL rule for allowed IPs to SSH port"
+    parameter_name = "ssh_port"
+    parameter_type = "numeric"
+	}
 }
 ```
 

--- a/website/docs/r/scc_profile_attachment.html.markdown
+++ b/website/docs/r/scc_profile_attachment.html.markdown
@@ -74,10 +74,10 @@ Nested schema for **notifications**:
 	* `enabled` - (Boolean) enabled notifications.
 * `attachment_parameters` - (List) The request payload of the attachment parameters.
 Nested schema for **attachment_parameters**:
-    * `parameter_name` - (String) The name of the parameter to target.
-    * `parameter_display_name` - (String) The display name of the parameter shown in the UI.
-    * `parameter_type` - (String) The type of the parameter value.
-    * `parameter_value` - (String) The value of the parameter.
+    * `parameter_name` - (Required, String) The name of the parameter to target.
+    * `parameter_display_name` - (Required, String) The display name of the parameter shown in the UI.
+    * `parameter_type` - (Required, String) The type of the parameter value.
+    * `parameter_value` - (Required, String) The value of the parameter.
     * `assessment_type` - (String) The type of assessment the parameter uses. 
 * `schedule` - (String) The schedule of an attachment evaluation.
   * Constraints: Allowable values are: `daily`, `every_7_days`, `every_30_days`.

